src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1102,12 +1102,62 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                 // Don't lower if we're not emitting or if there were errors. 
                 // Methods that had binding errors are considered too broken to be lowered reliably.
-                if (_moduleBeingBuiltOpt == null || hasErrors)
+                if (_moduleBeingBuiltOpt == null || hasErrors || isEmptySynthesizedStaticConstructor(processedInitializers.BoundInitializers))
                 {
                     _diagnostics.AddRange(actualDiagnostics);
                     return;
                 }
 
+                // We should only go through with emitting a synthesized static constructor if it would contain
+                // "meaningful" statements. If it only initializes fields to default values we will skip emitting it.
+                bool isEmptySynthesizedStaticConstructor(ImmutableArray<BoundInitializer> initializers)
+                {
+                    if (methodSymbol is SynthesizedStaticConstructor)
+                    {
+                        foreach (var initializer in initializers)
+                        {
```

Comment:
>{ [](start = 24, length = 1)

Consider extracting the loop body to a helper method and use `return initializers.Any(i => HasStatements(i));`.